python/mlc_chat/model/orion/orion_model.py

```diff
@@ -0,0 +1,400 @@
+                bold("context_window_size"),
+            )
+            self.prefill_chunk_size = self.context_window_size
+
+
+# pylint: disable=invalid-name,missing-docstring
+
+
+class OrionFFN(nn.Module):
+    def __init__(self, config: OrionConfig):
+        super().__init__()
+        self.intermediate_size = config.intermediate_size // config.tensor_parallel_shards
+        self.gate_up_proj = nn.Linear(
+            in_features=config.hidden_size,
+            out_features=2 * self.intermediate_size,
+            bias=False,
+        )
+        self.down_proj = nn.Linear(self.intermediate_size, config.hidden_size, bias=False)
+
+    def forward(self, x: Tensor):
+        concat_x1_x2 = self.gate_up_proj(x)
+        x1, x2 = op.split(concat_x1_x2, 2, axis=-1)
+        return self.down_proj(op.silu(x1) * x2)
+
+
+class OrionAttention(nn.Module):  # pylint: disable=too-many-instance-attributes
+    def __init__(self, config: OrionConfig):
+        self.head_dim = config.head_dim
+        self.num_q_heads = config.num_attention_heads // config.tensor_parallel_shards
+        assert (
+            config.num_key_value_heads % config.tensor_parallel_shards == 0
+        ), f"num_kv_heads({config.num_key_value_heads}) must be divisible by tensor_parallel_shards"
+        assert (
+            config.num_key_value_heads >= config.tensor_parallel_shards
+        ), f"Too large tensor_parallel_shards, must be smaller than {config.num_key_value_heads}"
+        self.num_kv_heads = config.num_key_value_heads // config.tensor_parallel_shards
+        self.qkv_proj = nn.Linear(
+            in_features=config.hidden_size,
+            out_features=(self.num_q_heads + 2 * self.num_kv_heads) * self.head_dim,
+            bias=False,
+        )
+        self.o_proj = nn.Linear(self.num_q_heads * self.head_dim, config.hidden_size, bias=False)
+
+    def forward(self, hidden_states: Tensor, paged_kv_cache: PagedKVCache, layer_id: int):
+        d, h_q, h_kv = self.head_dim, self.num_q_heads, self.num_kv_heads
+        b, s, _ = hidden_states.shape
+        # QKV Projection
+        qkv = self.qkv_proj(hidden_states)
+        qkv = op.reshape(qkv, (b, s, h_q + h_kv + h_kv, d))
+        # Attention
+        output = op.reshape(
+            paged_kv_cache.attention_with_fused_qkv(layer_id, qkv, self.num_q_heads),
+            (b, s, h_q * d),
+        )
+        return self.o_proj(output)
+
+    def batch_forward(self, hidden_states: Tensor, paged_kv_cache: PagedKVCache, layer_id: int):
+        d, h_q, h_kv = self.head_dim, self.num_q_heads, self.num_kv_heads
+        b, s, _ = hidden_states.shape
+        # QKV Projection
+        qkv = self.qkv_proj(hidden_states)
+        qkv = op.reshape(qkv, (b, s, h_q + h_kv + h_kv, d))
+        # Attention
+        output = op.reshape(
+            paged_kv_cache.attention_with_fused_qkv(layer_id, qkv, self.num_q_heads),
+            (b, s, h_q * d),
+        )
+        return self.o_proj(output)
```

Comment:
Hey, we had a PR that removed most of the `batch_forward` (but not all) functions in #1900. Would you like help check it out and remove most of the `batch_forward` in Orion?

Ideally, there should be only 4 occurrences of the word `batch_forward` in this file: one `def batch_forward` in `OrionForCasualLM`, and three `self.batch_forward` in `batch_prefill`, `batch_decode`, `batch_verify` respectively. All other `batch_forward` occurrences can be safely removed.